src/NeoSharp.Core/Messaging/Handlers/TransactionMessageHandler.cs

```diff
@@ -46,7 +46,8 @@ public TransactionMessageHandler(IBlockchain blockchain, ILogger<TransactionMess
             // Transaction is not added right away but queued to be verified and added.
             // It is the reason why we do not broadcast immediately.
 
-            var transactionAdded = await _blockchain.AddTransaction(transaction);
+            var transactionAdded = _blockchain.MemoryPool.Push(transaction);
```

Comment:
Please return this back. The handler does not care about MemoryPool. It just wants to push transaction using method on Blockchain. If the method delegates this task to MemoryPool inside it is ok.